### DISCUSSION TITLE
Error message reporting a dimension missmatch with values that are nowhere in the model.

### DISCUSSION THREAD

**Discussion by tiagoataquila** [onnx/onnx#3871] [Q&A]
Hello.  I am new to onnx and I am trying to  use it for inference with an unet model we developed with pytorch.

I save the torch model with 

`def export_onnx_model(model, input_shape, onnx_path, input_names=None, output_names=None, dynamic_axes=None):
    inputs = torch.ones(*input_shape)
    model(inputs)
    torch.onnx.export(model, inputs, onnx_path, input_names=input_names, output_names=output_names, dynamic_axes=dynamic_axes)

batch_size= 1
width =224
height=224
input_shape = (batch_size, 3, width, height)
onnx_path = "/content/gdrive/My Drive/COLAB_AI_TEAM/colab-resources/test_onnx.onnx"
input_names=['input']
output_names=['output']
dynamic_axes= {'input':{0:'batch_size' , 2:'width', 3:'height'}, 'output':{0:'batch_size' , 2:'width', 3:'height'}} #adding names for better debugging
export_onnx_model(model_a, input_shape, onnx_path, input_names, output_names, dynamic_axes)
`

It seems to work. Then I load and use it as:

`colored = cv2.cvtColor(img.astype(np.float32),cv2.COLOR_GRAY2RGB) 
colored_to_tensor = np.expand_dims(colored,axis=0)
colored_to_tensor = np.moveaxis(colored_to_tensor,-1,1)

onnx_inputs= {onnx_session.get_inputs()[0].name:colored_to_tensor}
onnx_output = onnx_session.run(None, onnx_inputs)
img_label = onnx_outputort_outs[0]`

The image is a rather large (3078, 2394)

but I get the following error messsage:

Fail: [ONNXRuntimeError] : 1 : FAIL : Non-zero status code returned while running Concat node. Name:'Concat_87' Status Message: concat.cc:156 PrepareForCompute Non concat axis dimensions must match: Axis 2 has mismatched dimensions of 193 and 194

The problem is, I cannot think from where did the numbers 193 and 194 arose. The expected result is of the same resolution of the original image (and works as such when I run the original torch model with the same image)

Can anyoen point me in the right direction?

**Comment by yuslepukhin** [comment]
Suggestion: use https://netron.app/ to visualize the model and trace node inputs outputs.
